plugins/org.jnario.spec/src/org/jnario/spec/scoping/SpecResourceDescriptionStrategy.xtend

```diff
@@ -13,18 +13,27 @@ import org.jnario.scoping.JnarioResourceDescriptionStrategy
 import org.jnario.spec.spec.ExampleGroup
 
 import static java.lang.String.*
+import org.eclipse.xtext.resource.IEObjectDescription
+import org.eclipse.xtext.common.types.JvmVoid
 
 class SpecResourceDescriptionStrategy extends JnarioResourceDescriptionStrategy {
 	
 	public static val ROOT_SPEC = "root"
 	public static val TRUE = "1"
 	public static val FALSE = "0"
+	public static val EXAMPLE_TYPE = "type"
 	
 	override createUserData(EObject eObject, ImmutableMap.Builder<String, String> userData) {
 		super.createUserData(eObject, userData);
 		if (eObject instanceof ExampleGroup){
 			val exampleGroup = eObject as ExampleGroup			
 			userData.put(ROOT_SPEC, valueOf(exampleGroup.isRoot))
+			if (exampleGroup.targetType !== null) {	
+				val targetType = exampleGroup.targetType
+				if (!(targetType.type instanceof JvmVoid)) {
+					userData.put(EXAMPLE_TYPE,exampleGroup.targetType.simpleName)					
```

Comment:
I'm not sure if we can safely call exampleGroup.getTargetType(). This has to sometimes return an unresolvable proxy? Otherwhise the original problem of the ExampleNameProvider wouldn't exist. 

I think the problem is that you write a spec:

MySpec.spec
```
describe MyClass{
}
```

MyClass.xtend
```
class MyClass{
}
```
So the ExampleGroup in MySpec.spec holds a reference targetType = MyClass,
I think the unresolvable problem happens, if MyClass was not generated yet, while you're indexing. E.g. in a cleanBuild.

I would rather move the NodeModel-approach to here, and remove it from the ExampleNameProvider. 
If we really want to make sure, we should also store the nodemodel in the storage (although it means a lot of more stuff)

I just checked, by default the node-model is not stored in the storage, but you can tell the storage facade to store the node-model as well. But its again a performance loss :( 
org.eclipse.xtext.resource.persistence.ResourceStorageFacade.setStoreNodeModel(boolean)